### PR TITLE
Upgrade Golang to 1.21

### PR DESCRIPTION
**Pull request by spolti**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/kserve/modelmesh-runtime-adapter
 
-go 1.19
+go 1.21
```
Comment (ckadner) [inline go.mod:3]: Did you run `go mod tidy -compat=1.21` to update the list of dependencies below? I would expect most of those to change.

**Review by ckadner — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (spolti) [reply go.mod:3]: as the rest-proxy one, just go mod tidy, let me try.

Comment (spolti) [reply go.mod:3]: here, we have some changes, pushed.
thanks.

**Review by ckadner — APPROVED** [review]
/lgtm

**Pull request merged**